src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionItemFilter.cs

```diff
@@ -0,0 +1,102 @@
+            FieldFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Fields, WellKnownTags.Field, 'f', builder);
+            EventFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Events, WellKnownTags.Event, 'v', builder);
+            PropertyFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Properties, WellKnownTags.Property, 'p', builder);
+            MethodFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Methods, WellKnownTags.Method, 'm', builder);
+            ExtensionMethodFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Extension_methods, WellKnownTags.ExtensionMethod, 'x', builder);
+            LocalAndParameterFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Locals_and_parameters, ImmutableArray.Create(WellKnownTags.Local, WellKnownTags.Parameter), 'l', builder);
+            KeywordFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Keywords, ImmutableArray.Create(WellKnownTags.Keyword), 'k', builder);
+            SnippetFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Snippets, ImmutableArray.Create(WellKnownTags.Snippet), 't', builder);
+            TargetTypedFilter = CreateCompletionFilterAndAddToBuilder(FeaturesResources.Target_type_matches, ImmutableArray.Create(WellKnownTags.TargetTypeMatch), 'j', builder);
+
+            allFilters = builder.ToImmutable();
+        }
+
+        private static CompletionFilter CreateCompletionFilter(
+            string displayText, ImmutableArray<string> tags, char accessKey)
+        {
+            var imageId = tags.GetFirstGlyph().GetImageId();
+            return new CompletionFilter(
+                displayText,
+                accessKey.ToString(),
+                new ImageElement(new ImageId(imageId.Guid, imageId.Id), EditorFeaturesResources.Filter_image_element));
+        }
+
+        private static CompletionFilter CreateCompletionFilterAndAddToBuilder(
+            string displayText, ImmutableArray<string> tags, char accessKey,
+            ImmutableDictionary<string, CompletionFilter>.Builder builder)
+        {
+            var filter = CreateCompletionFilter(displayText, tags, accessKey);
+            foreach (var tag in tags)
+            {
+                builder.Add(tag, filter);
+            }
+
+            return filter;
+        }
+
+        private static CompletionFilter CreateCompletionFilterAndAddToBuilder(
+            string displayText, string tag, char accessKey,
+            ImmutableDictionary<string, CompletionFilter>.Builder builder)
+            => CreateCompletionFilterAndAddToBuilder(displayText, ImmutableArray.Create(tag), accessKey, builder);
+
+        public static ImmutableArray<CompletionFilter> GetFilters(RoslynCompletionItem item)
+        {
+            var result = ArrayBuilder<CompletionFilter>.GetInstance();
```

Comment:
creating/freeing `ArrayBuilder` can now be done using using statement, just in case you are not aware of this change :)

e.g.
[http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs,76](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs,76)